File: api/Models/Lead.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using api.Enums;

namespace api.Models
{
    public class Lead
    {
        public int Id { get; set; }
        public string ContactFirstName { get; set; } = string.Empty;
        public string ContactLastName { get; set; } = string.Empty;
        public string ContactPhoneNumber { get; set; } = string.Empty;
        public string ContactEmail { get; set; } = string.Empty;
        public DateTime DateCreated { get; set; } = DateTime.UtcNow;
        public string Suburb { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }
        public LeadStatus Status { get; set; } = LeadStatus.Invited;
    }
}

File: frontend/src/types/Lead.ts
export enum LeadStatus {
    Invited = 0,
    Accepted = 1,
    Declined = 2,
  }
  
  export interface Lead {
    id: number;
    firstName: string;
    fullName?: string;
    phone?: string;
    email?: string;
    dateCreated: string;
    suburb: string;
    category: string;
    description: string;
    price: number;
    status: LeadStatus;
  }
  